### PR TITLE
Natspec added to parameters and random manager

### PR DESCRIPTION
**Pull request by SamAg19**

### CODE REVIEW

File: contracts/Core/RewardManager.sol [hunk 1]
```diff
@@ -21,7 +21,8 @@ contract RewardManager is Initializable, Constants, RewardManagerParams, IReward
     IBlockManager public blockManager;
     ICollectionManager public collectionManager;
 
-    /** @param stakeManagerAddress The address of the VoteManager contract
+    /**
+     * @param stakeManagerAddress The address of the StakeManager contract
```
Comment (abhishekvispute) [inline contracts/Core/RewardManager.sol:25]: hey here collection manger also needs to be added as param

Comment (abhishekvispute) [reply contracts/Core/RewardManager.sol:25]: also please write Natspec for internal functions as well

Comment (SamAg19) [reply contracts/Core/RewardManager.sol:25]: I have not added natspec on internal functions as it is not going to be parsed by the solidity compiler. It is still upto us though whether we want to add natspec to internal functions

Comment (SkandaBhat) [reply contracts/Core/RewardManager.sol:25]: Lets add them. Will be better for readability and auditing.

File: contracts/Core/RewardManager.sol [hunk 2]
```diff
@@ -81,6 +88,12 @@ contract RewardManager is Initializable, Constants, RewardManagerParams, IReward
         }
     }
 
+    /**
+     * @dev Penalties are given to stakers based their activity if they have been inactive for more than the grace period
+     * and their votes in the previous epoch compared to the medians confirmed. Penalties on botes depend upon how far were
```
Comment (abhishekvispute) [inline contracts/Core/RewardManager.sol:93]: I don't want to add review on English 😄 
But here it should be ``votes`` instead of ``botes``

**Review by abhishekvispute — APPROVED** [review]

File: contracts/Core/RewardManager.sol (hunk 2 above)
Comment (SamAg19) [reply contracts/Core/RewardManager.sol]: fixed https://github.com/razor-network/contracts/pull/712/commits/3300506664f50da270755398db4cd653542c4b21

**Review by abhishekvispute — APPROVED** [review]

**Review by hrishikeshio — APPROVED** [review]

**Comment by hrishikeshio** [comment]
what about failing gascompare? @SamAg19

**Pull request merged**